src/Symfony/Component/ObjectMapper/ObjectMapper.php

```diff
@@ -0,0 +1,312 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ObjectMapper;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\ObjectMapper\Exception\MappingException;
+use Symfony\Component\ObjectMapper\Exception\MappingTransformException;
+use Symfony\Component\ObjectMapper\Metadata\Mapping;
+use Symfony\Component\ObjectMapper\Metadata\ObjectMapperMetadataFactoryInterface;
+use Symfony\Component\ObjectMapper\Metadata\ReflectionObjectMapperMetadataFactory;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+
+/**
+ * Object to object mapper.
+ *
+ * @experimental
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class ObjectMapper implements ObjectMapperInterface
+{
+    /**
+     * An SplObjectStorage that tracks recursive references.
+     */
+    private ?\SplObjectStorage $objectMap = null;
+
+    /**
+     * @param ContainerInterface<TransformCallableInterface> $transformCallableLocator
+     * @param ContainerInterface<ConditionCallableInterface> $conditionCallableLocator
```

Comment:
As notified by psalm, the `Psr\Container\ContainerInterface` doesn't have a template param. I don't think the PR to add it will be accepted: https://github.com/php-fig/container/pull/50

```suggestion
     * @param ContainerInterface $transformCallableLocator
     * @param ContainerInterface $conditionCallableLocator
```

Even [Symfony's ContainerInterface](https://github.com/symfony/symfony/blob/7.3/src/Symfony/Component/DependencyInjection/ContainerInterface.php) does not have this template param.